File: 지현배/2205/220530/2023.py
import sys
input = sys.stdin.readline

N = int(input())

prime = {
    1: [2, 3, 5, 7]
}

def check(value):
    for i in range(2, int(value ** 0.5) + 1):
        if value % i == 0:
            return False
    return True

for i in range(2, N + 1):
    prime[i] = []
    for p in prime[i - 1]:
        for j in range(5):
            if check(10 * p + 2 * j + 1):
                prime[i].append(10 * p + 2 * j + 1)

print(*prime[N], sep='\n')

File: 지현배/2205/220530/2023.js
const N = +require("fs")
  .readFileSync(require("path").join(__dirname, "dev/stdin"))
  .toString()
  .trim();

const prime = {
  1: [2, 3, 5, 7],
};

function check(value) {
  for (let i = 2; i <= value ** 0.5; i++) {
    if (value % i === 0) {
      return false;
    }
  }
  return true;
}

for (let i = 2; i < N + 1; i++) {
  prime[i] = [];
  prime[i - 1].forEach((p) => {
    for (let j = 0; j < 5; j++) {
      if (check(10 * p + 2 * j + 1)) {
        prime[i].push(10 * p + 2 * j + 1);
      }
    }
  });
}

console.log(prime[N].join("\n"));
